slasher/rpc/detect_update_min_max_span.go

```diff
@@ -0,0 +1,93 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// DetectAndUpdateMaxSpan is used to detect and update the max span of an incoming attestation.
+// logic is following the detection method designed by https://github.com/protolambda
+// from here: https://github.com/protolambda/eth2-surround/blob/master/README.md#min-max-surround
+func (ss *Server) DetectAndUpdateMaxSpan(ctx context.Context, source uint64, target uint64, validatorIdx uint64) (surroundedTargetEpoch uint64, err error) {
+	span := target - source + 1
+	if span > params.BeaconConfig().WeakSubjectivityPeriod {
+		return 0, fmt.Errorf("attestation detection supports only weak subjectivity period: %v target - source: %v > weakSubjectivityPeriod", params.BeaconConfig().WeakSubjectivityPeriod, span)
```

Comment:
Line too long, make multiline as:
```suggestion
		return 0, fmt.Errorf("
		    %v target - source: %v > weakSubjectivityPeriod", 
		    params.BeaconConfig().WeakSubjectivityPeriod, 
		    span,
		)
```